src/lib.rs

```diff
@@ -454,6 +627,75 @@ impl<T: ToMutByteSlice, U: AsMut<[T]> + ?Sized> AsMutByteSlice<T> for U {
     }
 }
 
+/// Trait for converting from a byte vec to a vec of a fundamental, built-in
+/// numeric type.
+///
+/// # Example
+/// ```
+/// # extern crate byte_slice_cast;
+/// use byte_slice_cast::*;
+///
+/// # fn main() {
+/// let mut vec: Vec<u8> = vec![1u8, 2u8, 3u8, 4u8, 5u8, 6u8];
+/// let converted_vec = vec.into_vec_of::<u16>().unwrap();
+///
+/// if cfg!(target_endian = "big") {
+///     assert_eq!(converted_vec, vec![0x0102, 0x0304, 0x0506]);
+/// } else {
+///     assert_eq!(converted_vec, vec![0x0201, 0x0403, 0x0605]);
+/// }
+/// # }
+/// ```
+
+pub trait IntoVecOf {
+    fn into_vec_of<T: FromByteVec>(self) -> Result<Vec<T>, Error>;
+}
+
+impl IntoVecOf for Vec<u8> {
```

Comment:
This should probably be inside the macro and impl'd for all the types

Reply:
So, `IntoVecOf` is the reciprocal of `FromByteVec` both are converting from `u8` to other numerical types, updating `IntoVecOf` means we would also need to make `FromByteVec` generic as well, right?

```rust
        unsafe impl FromByteVec for $to {
            fn from_byte_vec(mut vec: Vec<u8>) -> Result<Vec<$to>, Error> {
```

Reply:
How would it work otherwise? We need a trait that goes from `Vec<T>` to `Vec<u8>` and the inverse

Maybe I'm missing something :)

Reply:
There is a trait `IntoByteVec` which is implemented for `Vec<T>` so it would allow it to convert any type to `Vec<u8>` and `IntoVecOf` is implemented for `Vec<u8>` it would convert it into any `Vec<T>`

Reply:
But there is no direct conversion like `Vec<16>` to `Vec<32>`.

Reply:
Ah! There's no need for direct conversion, we don't have that for `[T]` either. And we wouldn't need a new trait for that either, it can be implemented via the existing ones (first to bytes and then to the target).

So that's all fine then. The direct conversions can be added in a separate PR if someone needs them or wants to implement them :)